app/assets/javascripts/notebook/magic/tsChart.coffee

```diff
@@ -11,8 +11,13 @@ define([
     chart_container = $("<div>").addClass("custom-c3-chart").attr("width", w+"px").attr("height", h+"px")
     chart_container.attr("id", "custom-c3-chart-"+@genId).appendTo(container)
 
-    data = {
-    }
+    data = {}
+
+    # Datetime data are coming as formatted strings. We need to tell about the format to c3 lib. But we still want
+    # to support opening old notebooks with old format (timestamp in miliseconds) without need to regenerate the data.
+    if (@dataInit.length > 0) && typeof @dataInit[0][options.x] is 'string'
```

Comment:
I added a tiny commit. I'd say `(dataInit.length >0)` should be cleaner than dealing with `undefined` (as before). 

@james64 if you can test/confirm this works (it should), I'll merge it right away.

Reply:
Javascript is new world for me so I am happy to be advised on things like this. I have tested it and it works. :+1: